Content.Shared/Sound/Components/EmitSoundOnInteractUsingComponent.cs

```diff
@@ -0,0 +1,13 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Sound.Components;
+
+/// <summary>
+/// Whenever this item is used upon by a specific entity prototype in the hand of a user, play a sound
+/// </summary>
+[RegisterComponent]
+public sealed partial class EmitSoundOnInteractUsingComponent : BaseEmitSoundComponent
+{
+    [DataField("UsedItemProto", false, 1, true)]
+    public ProtoId<EntityPrototype> UsedItemProto = new();
```

Comment:
ok so general tips:
1. default tag would be usedItemProto which is the style, do not override it with PascalCase
2. random bools ints etc should be labelled e.g. `serverOnly: true` so its obvious what they are, right now i have to guess what they are
3. probably use string.Empty instead of new() (not sure what ProtoId() constructor is and if it does that already)

in line with sloths review, for this to make it not use a prototype id (entities can be made without prototypes and other edge cases that this would fail on, besides being non reusable) use `[DataField(required: true)] public EntityWhitelist Whitelist = new();` then have `whitelist` in yml either have a tag or component that the target should have. this means it can be reused for other things that should have the gavel sound etc. instead of only working for a single prototype.